packages/frontend/src/pages/Playground.svelte

```diff
@@ -1,5 +1,151 @@
 <script lang="ts">
+import { conversations } from '../stores/conversations';
+import { studioClient } from '/@/utils/client';
+import {
+  isAssistantChat,
+  type ChatMessage,
+  type UserChat,
+  isPendingChat,
+  isUserChat,
+  type AssistantChat,
+} from '@shared/src/models/IPlaygroundMessage';
+import NavPage from '../lib/NavPage.svelte';
+import { playgrounds } from '../stores/playgrounds-v2';
+import { catalog } from '../stores/catalog';
+import Button from '../lib/button/Button.svelte';
+import { afterUpdate } from 'svelte';
+
 export let playgroundId: string;
+let prompt: string;
+let sendEnabled = false;
+let scrollable: Element;
+let lastIsUserMessage = false;
+let errorMsg = '';
+
+$: conversation = $conversations.find(conversation => conversation.id === playgroundId);
+$: playground = $playgrounds.find(playground => playground.id === playgroundId);
+$: model = $catalog.models.find(model => model.id === playground?.modelId);
+$: {
+  if (conversation?.messages.length) {
+    const latest = conversation.messages[conversation.messages.length - 1];
+    if (isAssistantChat(latest) && !isPendingChat(latest)) {
+      sendEnabled = true;
+    }
+    lastIsUserMessage = isUserChat(latest);
+  } else {
+    sendEnabled = true;
+  }
+}
+
+const roleNames = {
+  system: 'System',
+  user: 'User',
+  assistant: 'Assistant',
+};
+
+function getMessageParagraphs(message: ChatMessage): string[] {
+  if (isAssistantChat(message)) {
+    if (!isPendingChat(message)) {
+      return message.content?.split('\n') ?? [];
+    } else {
+      return message.choices
+        .map(choice => choice.content)
+        .join('')
+        .split('\n');
+    }
+  } else if (isUserChat(message)) {
+    const msg = message as UserChat;
+    return msg.content?.split('\n') ?? [];
+  }
+  return [];
+}
+
+function askPlayground() {
+  studioClient
```

Comment:
When you click on the submit, it should be displayed or show a loading state. It has a `inProgress` properties.

Also the error state should be reset when clicking again on the button